python/ray/serve/schema.py

```diff
@@ -487,6 +517,118 @@ def get_empty_schema_dict() -> Dict:
         return {"applications": []}
 
 
+@PublicAPI(stability="alpha")
+class DeploymentDetails(BaseModel, extra=Extra.forbid):
+    name: str = Field(description="Deployment name.")
```

Comment:
It seems like by default name can be empty string. Can we make the default application name "default" or something?

The reason is I'm hoping to use this field as a unique identifier for each application and put it as part of the url for the UI.

`/serve/applications/:name`. If name is empty string, the url will be very weird.

Reply:
also, i need a cell to serve as a good link from the table view to the detail page. The name column makes the most sense, but it's weird if it's an empty string.

<img width="1700" alt="Screenshot 2023-03-11 at 1 26 49 PM" src="https://user-images.githubusercontent.com/711935/224512270-1c0c847b-610d-4c89-ad59-c76425c47167.png">

Reply:
Yeah, this is a known issue. I made an attempt to change it but turns out it's not a very simple change, due to backwards compatibility issues. We decided not to make it a blocker before feature completion of multi-app REST/CLI support, but we plan on addressing this soon after.

The problem is if users use the old, single-app API, the default name for their single app will be "", so it will actually be a common case. Would it be possible for the dashboard to show the name column as something explicit like `<empty>`?

Reply:
Actually, we should not allow users to mix single-app and multi-app API to begin with; in particular if users use single-app API, maybe we shouldn't even show the new dashboard metrics, since it fetches from the new endpoint /api/serve/applications/? @sihanwang41 @edoakes @shrekris-anyscale

Reply:
> in particular if users use single-app API, maybe we shouldn't even show the new dashboard metrics, since it fetches from the new endpoint /api/serve/applications/

I think it's still a good idea to show dashboard metrics for the single-app use-case. Otherwise, Serve would effectively be deprecating single-app configs in the next release since users would lose a lot of existing observability.

Reply:
Yup, that's true. @alanwguo On your end, this means (currently) the name will often be the empty string. Is it possible to treat that as a special case and display something explicit?